onnxruntime/python/tools/transformers/benchmark_gpt2.py

```diff
@@ -79,44 +120,84 @@ def onnxruntime_inference(ort_session, input_ids, past=None, attention_mask=None
     return ort_outputs, average_latency
 
 
-def onnxruntime_inference_with_binded_io(ort_session,
-                                         input_ids,
-                                         output_buffers,
-                                         max_last_state_size,
-                                         last_state_shape,
-                                         past=None,
-                                         attention_mask=None,
-                                         max_present_size=None,
-                                         present_shape=None,
-                                         total_runs=100):
+def get_dummy_inputs(batch_size, past_sequence_length, num_attention_heads, hidden_size, num_layer, vocab_size, device,
+                     use_attention_mask, float16):
+    float_type = torch.float16 if float16 else torch.float32
+    past_shape = [2, batch_size, num_attention_heads, past_sequence_length, int(hidden_size / num_attention_heads)]
+    dummy_past = [torch.rand(past_shape, dtype=float_type, device=device) for _ in range(num_layer)]
+    dummy_input_ids = torch.randint(low=0, high=vocab_size - 1, size=(batch_size, 1), dtype=torch.int64, device=device)
+    if use_attention_mask:
+        dummy_attention_mask = torch.ones([batch_size, 1], dtype=float_type, device=device)
+        dummy_position_ids = torch.ones([batch_size, 1], dtype=torch.int64, device=device) * past_sequence_length
+        return dummy_input_ids, dummy_past, dummy_attention_mask, dummy_position_ids
+
+    return dummy_input_ids, dummy_past, None, None
+
+
+def get_output_shapes(batch_size, past_sequence_length, config, use_LMHead):
+    num_attention_heads = config.num_attention_heads
+    hidden_size = config.hidden_size
+    num_layer = config.n_layer
+    vocab_size = config.vocab_size
+
+    last_state_shape = [batch_size, 1, vocab_size] if use_LMHead else [batch_size, 1, hidden_size]
+    present_state_shape = [
+        2, batch_size, num_attention_heads, past_sequence_length + 1,
+        int(hidden_size / num_attention_heads)
+    ]
+
+    output_shapes = {}
+    last_state_name = "prediction_scores" if use_LMHead else "last_state"
```

Comment:
I wonder if these input/output names are always the same across models. We can get the names from ort session.get_inputs()/get_outputs()[i].name instead of hard coding the names.

Reply:
This benchmark script only deal with two models (GPTModel and GPTLMHeadModel), so we hard code the names  and shape info.

If there are many models (like benchmark.py), we might use ort session and dummy inputs to get an example output, so we know the output names and shapes.